attribution-data-file-share/src/main/java/gov/cms/ab2d/attributionDataShare/AttributionDataShare.java

```diff
@@ -0,0 +1,94 @@
+package gov.cms.ab2d.attributionDataShare;
+
+import gov.cms.ab2d.lambdalibs.lib.PropertiesUtil;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.LambdaLogger;
+
+public class AttributionDataShare {
+    
+    private static final String filePartialName = "ab2d-beneids_";
+    private static final String fileFormat = ".txt";
+
+    private static final String filePath = "/opt/";
+
+    private static final String SELECT_ALL_FROM_COVERAGE = "SELECT beneficiary_id from public.coverage";
+
+    // Returns a string with the fileFullPath to the file we wrote out.
+    // I.E: "ab2d-beneids_2023-08-16T12:08:56.235-0700.txt"
+    public String handleRequest(Context context) throws IOException {
+        LambdaLogger logger = context.getLogger();
+        logger.log("AttributionDataShare Lambda is started");
+
+        String currentDate = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ").format(new Date());
+
+        String fileFullName = filePartialName + currentDate + fileFormat;
+        
+        String fileFullPath = filePath + fileFullName;
+
+        try {
+            Connection dbConnection = getConnection();
+            List<String> coverageDataList = fetchCoverageData(dbConnection, SELECT_ALL_FROM_COVERAGE);
+
+            FileWriter fileWriter = new FileWriter(fileFullPath);
```

Comment:
if you initialize `fileWriter` inside the `try`, it will be unaccessible in the `finaly`.
You need to move ` FileWriter fileWriter = new FileWriter(fileFullPath);` upper before `try`

Reply:
I see in the commit you pushed, you placed the fileWriter initialization inside of the try.

Like `try(Filewriter fileWriter = new FileWriter(fileFullPath) { ... `

I have never seen someone pass arguments to try before, so that is neat. Not saying this is wrong, but just wondering if this is a good practice or if we should just do it inside the try?

Like `try() {
            Filewriter fileWriter = new FileWriter(fileFullPath);
            ...
         }`

Reply:
`try(Filewriter fileWriter = new FileWriter(fileFullPath) {`
 equals to
`Filewriter fileWriter = new FileWriter(fileFullPath);
 try {`
 
 Actually, I'd like to ask to replace FileWriter to BufferedWriter
https://stackoverflow.com/questions/12350248/java-difference-between-filewriter-and-bufferedwriter
it should be more efficient in out case

Reply:
Done. Please double check that I have implemented this properly. Do we still need the fileWriter at all? It looks like BufferedWriter takes in a fileWriter so I think we do but I just wanted to double check.